### PR TITLE
Add status changing tests to partner request spec

### PR DESCRIPTION
**Pull request by chaserx**
### Description

This increases the test coverage for the Partner API. Specifically, it adds tests for each status change. 

There's also some minor cleanup to the routes to reflect how we're using the routes currently and to the response that we provide on partner update. 

### Type of change

* New feature (non-breaking change which adds functionality)

### How Has This Been Tested?

* tests run/pass locally
* CI
* **should verify that the client applications still work**

### CODE REVIEW

File: spec/requests/api/v1/partners_requests_spec.rb [hunk 1]
```diff
@@ -48,6 +48,85 @@
...
+        expect(partner.reload.partner_status).to eq("Verified")
+      end
+
+      it 'shows verified status in the response body' do
+        valid_partner_update_request(params, headers)
+        expect(JSON.parse(response.body)['message']).to eq('Partner status: Verified.')
+      end
+    end
+
+    context "when we try to set the partner to blarg" do
```
Comment (benreyn) [inline spec/requests/api/v1/partners_requests_spec.rb]: Is this the behavior that we want to be testing? I think ultimately we will want to have this return some errors, but that is surely out of scope here. (probably v2 of the API)

File: spec/requests/api/v1/partners_requests_spec.rb [hunk 2]
```diff
@@ -48,6 +48,85 @@
...
+      let(:params) { { partner: { diaper_partner_id: partner.diaper_bank_id, status: "approved" }} }
+
+      it 'returns OK' do
+        valid_partner_update_request(params, headers)
+        expect(response).to have_http_status(:ok)
+      end
+
+      it 'sets the status to Verified' do
+        valid_partner_update_request(params, headers)
+        expect(partner.reload.partner_status).to eq("Verified")
```
Comment (benreyn) [inline spec/requests/api/v1/partners_requests_spec.rb]: I think @kdreyeroren Is doing some work to ensure that we are using consistent `"snake_case"` strings for partner_status. We can address as a merge conflict later.

**Comment by benreyn** [comment]
Thank you @chaserx! Ill merge when tests are green. 👍

File: spec/requests/api/v1/partners_requests_spec.rb (hunk 1 above)
Comment (chaserx) [reply spec/requests/api/v1/partners_requests_spec.rb]: yes. I think we need to change this in the future. i wanted to only change one thing at time.

File: spec/requests/api/v1/partners_requests_spec.rb (hunk 2 above)
Comment (chaserx) [reply spec/requests/api/v1/partners_requests_spec.rb]: ok

**Pull request merged**